File: program/find-the-maximum-sum-of-a-contiguous-subarray-(kadane's-algorithm)/find-the-maximum-sum-of-a-contiguous-subarray-(kadane's-algorithm).py
# Python program to find maximum contiguous subarray

# Function to find the maximum contiguous subarray
def maxSubArraySum(a, size):

	max_so_far = a[0]
	max_ending_here = a[0]

	for i in range(1, size):
		max_ending_here = max_ending_here + a[i]
		if (max_so_far < max_ending_here):
			max_so_far = max_ending_here

		if max_ending_here < 0:
			max_ending_here = 0
	return max_so_far

# Driver function to check the above function


a = [-2, -3, 4, -1, -2, 1, 5, -3]

print("Maximum contiguous sum is", maxSubArraySum(a, len(a)))

File: program/find-the-maximum-sum-of-a-contiguous-subarray-(kadane's-algorithm)/FindTheMaximumSumOfAContiguousSubarray(kadane'sAlgorithm).java
public class FindTheMaximumSumOfAContiguousSubarray{
//(kadane'sAlgorithm)
    public static void main(String[] args) {
    int[] numberArray = {-2, 1, 9, 4, -1, 2, -2, -5, 4};
    System.out.println("Maximum contiguous sum is "
      +  maxSubArraySum(numberArray));
  }
  static int maxSubArraySum(int numberArray[]) 
    { 
        int max_so_far = numberArray[0]; 
        int curr_max = numberArray[0]; 
  
        for (int i = 1; i < numberArray.length; i++) 
        { 
            curr_max = Math.max(numberArray[i], curr_max+numberArray[i]); 
            max_so_far = Math.max(max_so_far, curr_max); 
        } 
        return max_so_far; 
    } 
}
 

